addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AmazonEchoControlHandlerFactory.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.amazonechocontrol.internal;
+
+import static org.openhab.binding.amazonechocontrol.AmazonEchoControlBindingConstants.*;
+
+import java.util.Hashtable;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.amazonechocontrol.handler.AccountHandler;
+import org.openhab.binding.amazonechocontrol.handler.EchoHandler;
+import org.openhab.binding.amazonechocontrol.handler.FlashBriefingProfileHandler;
+import org.openhab.binding.amazonechocontrol.internal.discovery.AmazonEchoDiscovery;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.http.HttpService;
+
+/**
+ * The {@link AmazonEchoControlHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Michael Geramb - Initial contribution
+ */
+@Component(service = ThingHandlerFactory.class, immediate = true, configurationPid = "binding.amazonechocontrol")
+@NonNullByDefault
+public class AmazonEchoControlHandlerFactory extends BaseThingHandlerFactory {
+
+    @Nullable
+    HttpService httpService;
+    @Nullable
+    AmazonEchoDiscovery amazonEchoDiscovery;
+
+    boolean showIdsInGUI;
+    @Nullable
+    ServiceRegistration<?> discoverServiceRegistration;

```

Comment:
You actually need to keep a Map of service registrations, see the Hue binding as the example that I had referred to before.
Currently, you have a single discovery service instance for your whole system. Instead, a discovery service instance should be associated to an account/bridge. So if you setup 3 Amazon accounts, you should also have 3 discovery services in place.

Reply:
You do not seem to have addressed this comment yet.

Reply:
Sorry, I just noticed that you did - you only left [this line](https://github.com/openhab/openhab2-addons/pull/3087/files#r190286713), which confused me.